src/main/java/appeng/parts/reporting/PartThroughputMonitor.java

```diff
@@ -0,0 +1,205 @@
+package appeng.parts.reporting;
+
+import java.io.IOException;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.Vec3;
+
+import org.lwjgl.opengl.GL11;
+
+import appeng.api.networking.IGridNode;
+import appeng.api.networking.ticking.IGridTickable;
+import appeng.api.networking.ticking.TickRateModulation;
+import appeng.api.networking.ticking.TickingRequest;
+import appeng.api.storage.data.IAEItemStack;
+import appeng.client.texture.CableBusTextures;
+import appeng.core.settings.TickRates;
+import appeng.helpers.Reflected;
+import appeng.util.IWideReadableNumberConverter;
+import appeng.util.Platform;
+import appeng.util.ReadableNumberConverter;
+import io.netty.buffer.ByteBuf;
+
+/**
+ * @author MCTBL
+ * @version rv3-beta-538-GTNH
+ * @since rv3-beta-538-GTNH
+ */
+public class PartThroughputMonitor extends AbstractPartMonitor implements IGridTickable {
+
+    private static final IWideReadableNumberConverter NUMBER_CONVERTER = ReadableNumberConverter.INSTANCE;
+
+    private static final CableBusTextures FRONT_BRIGHT_ICON = CableBusTextures.PartThroughputMonitor_Bright;
+    private static final CableBusTextures FRONT_DARK_ICON = CableBusTextures.PartThroughputMonitor_Dark;
+    private static final CableBusTextures FRONT_COLORED_ICON = CableBusTextures.PartThroughputMonitor_Colored;
+    private static final CableBusTextures FRONT_COLORED_ICON_LOCKED = CableBusTextures.PartThroughputMonitor_Dark_Locked;
+
+    private static final String[] TIME_UNIT = { "/t", "/s", "/m", "/h" };
+    private static final float[] NUMBER_MULTIPLIER = { 1, 20, 1_200, 72_000 };
```

Comment:
Please, organize all these values in nested class/enum (for better readability and maintainability). It's much easier to add/modify single enum constant than searching through multiple arrays.